packages/data/src/components/with-dispatch/test/index.js

```diff
@@ -10,224 +11,138 @@ import withDispatch from '../';
 import { createRegistry } from '../../../registry';
 import { RegistryProvider } from '../../registry-provider';
 
-describe( 'withDispatch', () => {
-	let registry;
-	beforeEach( () => {
-		registry = createRegistry();
-	} );
+jest.useRealTimers();
 
-	it( 'passes the relevant data to the component', () => {
-		const store = registry.registerStore( 'counter', {
-			reducer: ( state = 0, action ) => {
-				if ( action.type === 'increment' ) {
-					return state + action.count;
-				}
-				return state;
-			},
-			actions: {
-				increment: ( count = 1 ) => ( { type: 'increment', count } ),
-			},
-		} );
-
-		const Component = withDispatch( ( _dispatch, ownProps ) => {
+describe( 'withDispatch', () => {
+	const storeOptions = {
+		reducer: ( state = 0, action ) => {
+			if ( action.type === 'inc' ) {
+				return state + action.count;
+			}
+			return state;
+		},
+		actions: {
+			increment: ( count = 1 ) => ( { type: 'inc', count } ),
+		},
+		selectors: {
+			getCount: ( state ) => state,
+		},
+	};
+
+	it( 'passes the relevant data to the component', async () => {
+		const user = userEvent.setup();
+		const registry = createRegistry();
+		registry.registerStore( 'counter', storeOptions );
+
+		const Component = withDispatch( ( dispatch, ownProps ) => {
 			const { count } = ownProps;
 
 			return {
 				increment: () => {
-					const actionReturnedFromDispatch = Promise.resolve(
-						_dispatch( 'counter' ).increment( count )
-					);
-					return expect(
-						actionReturnedFromDispatch
-					).resolves.toEqual( {
-						type: 'increment',
-						count,
-					} );
```

Comment:
This is not a bad thing to check: that the `increment` function prop injected to `Component` has the right return value. In case of a plain action, it's a promise of the action object. Would be nice to keep the check.

I would put it to the user-land code, to the `button` `onClick` handler:
```js
async function onClick() {
  const result = await props.increment();
  expect( result ).toEqual( ... );
}
```